### PR TITLE
Allow admin to manually change roles of user

### PR DESCRIPTION
**Pull request by AndyLodestar**
Allow admin to manually change roles of user

### CODE REVIEW

**Comment by AndyLodestar** [comment]
Update role order

File: ElastosBountyProgram/back-end/src/service/UserService.ts [hunk 1]
```diff
@@ -132,16 +132,23 @@ export default class extends Base {
 
         let user = await db_user.findById(userId)
         let countryChanged = false
+        let is_admin = this.currentUser.role === constant.USER_ROLE.ADMIN ? true : false
```
Comment (WiktorStarczewski) [inline ElastosBountyProgram/back-end/src/service/UserService.ts]: ```const is_admin = this.currentUser.role === constant.USER_ROLE.ADMIN```

File: ElastosBountyProgram/back-end/src/service/UserService.ts [hunk 2]
```diff
@@ -132,16 +132,23 @@ export default class extends Base {
...
-            this.currentUser._id.toString() !== userId)) {
+        if (!this.currentUser || (!is_admin && this.currentUser._id.toString() !== userId)) {
             throw 'Access Denied'
         }
 
         if (!user) {
             throw `userId: ${userId} not found`
         }
 
+        if(is_admin){
```
Comment (WiktorStarczewski) [inline ElastosBountyProgram/back-end/src/service/UserService.ts]: ```if (is_admin && param.role) {```

File: ElastosBountyProgram/front-end/src/config/data.js [hunk 3]
```diff
@@ -375,4 +375,9 @@ export default {
         female: 'Female',
         other: 'Other',
     },
+    mappingRoleToName: {
+        MEMBER: 'Member',
```
Comment (WiktorStarczewski) [inline ElastosBountyProgram/front-end/src/config/data.js]: To be used as display values, those should be in the I18N module, and accessed from there for translation purposes. (consider ```I18N.get('key')```)

File: ElastosBountyProgram/front-end/src/module/form/UserEditForm/Component.js [hunk 4]
```diff
@@ -116,6 +116,23 @@ class C extends BaseComponent {
         const username_el = (
             <Input size="large" disabled/>
         )
+        
+        const role_fn = getFieldDecorator('role', {
+            rules: [{required: true, message: 'Please select your role'}],
```
Comment (WiktorStarczewski) [inline ElastosBountyProgram/front-end/src/module/form/UserEditForm/Component.js]: 1. Please select a role (not yours, the admin potentially edits someone elses account)
2. Please move that string to I18N

File: ElastosBountyProgram/front-end/src/module/form/UserEditForm/Component.js [hunk 5]
```diff
@@ -398,6 +416,11 @@ class C extends BaseComponent {
                         <FormItem label="Username" {...formItemLayout}>
                             {p.username}
                         </FormItem>
+                        {this.props.is_admin && 
+                        <FormItem label="Role" {...formItemLayout}>{
+                            p.role}
```
Comment (WiktorStarczewski) [inline ElastosBountyProgram/front-end/src/module/form/UserEditForm/Component.js]: Indentation (move { to new line to make ```{p.role}```)

File: ElastosBountyProgram/front-end/src/service/UserService.js [hunk 6]
```diff
@@ -24,6 +24,8 @@ export default class extends BaseService {
 
         if ([USER_ROLE.ADMIN, USER_ROLE.COUNCIL].includes(res.user.role)) {
             await this.dispatch(userRedux.actions.is_admin_update(true))
+        }else{
+            await this.dispatch(userRedux.actions.is_admin_update(false))
         }
         if ([USER_ROLE.LEADER].includes(res.user.role)) {
             await this.dispatch(userRedux.actions.is_leader_update(true))
```
Comment (WiktorStarczewski) [inline ElastosBountyProgram/front-end/src/service/UserService.js]: Might as well repeat that else here
```
} else {
     await this.dispatch(userRedux.actions.is_leader_update(false))
}
```

**Comment by WiktorStarczewski** [comment]
Please fix this test also:
```
Tests for User › It should not be possible to change other users email

    AssertionError: expected [TypeError: Cannot read property 'role' of undefined] to equal 'Access Denied'

      52 |             assert.fail(`Should fail with ${expectedError}`)
      53 |         } catch (err) {
    > 54 |             expect(err).to.be.equal(expectedError)
         |                               ^
      55 |         }
      56 |     })
      57 | 

      at Object.<anonymous> (src/service/.test/User.test.ts:54:31)
          at Generator.throw (<anonymous>)
      at rejected (src/service/.test/User.test.ts:5:65)
```

**Comment by AndyLodestar** [comment]
All the above problems have been solved

**Comment by WiktorStarczewski** [comment]
Please rebase on top of staging again, some commits dont belong here.

**Comment by AndyLodestar** [comment]
I changed it back.

**Pull request merged**